### ISSUE TITLE
Default initializer for NetworkReachabilityManager doesn't seem to be conformant with Apple IPv6 Guidelines

### ISSUE THREAD

**Issue by soxjke** [Alamofire/Alamofire#1789]
I'm relying on my usage of ```NetworkReachabilityManager``` and rejection note from Apple:

> Next Steps
>
> Please revise your app and test it on a device while connected to an IPv6 network (all apps must support IPv6) to ensure that it runs as expected.
>
> Resources
>
> For information about supporting IPv6 Networks, please refer to Supporting IPv6 DNS64/NAT64 Networks and About Networking.

Referenced document [Supporting IPv6 DNS64/NAT64 Networks](https://developer.apple.com/library/mac/documentation/NetworkingInternetWeb/Conceptual/NetworkingOverview/UnderstandingandPreparingfortheIPv6Transition/UnderstandingandPreparingfortheIPv6Transition.html#//apple_ref/doc/uid/TP40010220-CH213-SW1) explicitly states, that:

> The Reachability APIs (see SCNetworkReachability Reference) are intended for diagnostic purposes after identifying a connectivity issue. Many apps incorrectly use these APIs to proactively check for an Internet connection by calling the SCNetworkReachabilityCreateWithAddress method and passing it an IPv4 address of 0.0.0.0, which indicates that there is a router on the network. However, the presence of a router doesn’t guarantee that an Internet connection exists. In general, avoid preflighting network reachability. Just try to make a connection and gracefully handle failures. If you must check for network availability, avoid calling the SCNetworkReachabilityCreateWithAddress method. Call the SCNetworkReachabilityCreateWithName method and pass it a hostname instead.
>
> Some apps also pass the SCNetworkReachabilityCreateWithAddress method an IPv4 address of 169.254.0.0, a self-assigned link-local address, to check for an active Wi-Fi connection. To check for Wi-Fi or cellular connectivity, look for the network reachability flag kSCNetworkReachabilityFlagsIsWWAN instead.

I'm using 3.4.2 version of Alamofire, and https://github.com/Alamofire/Alamofire/blob/3.4.2/Source/NetworkReachabilityManager.swift#L126 has initializer 

```
    public convenience init?() {
        var address = sockaddr_in()
        address.sin_len = UInt8(sizeofValue(address))
        address.sin_family = sa_family_t(AF_INET)

        guard let reachability = withUnsafePointer(&address, {
            SCNetworkReachabilityCreateWithAddress(nil, UnsafePointer($0))
        }) else { return nil }

        self.init(reachability: reachability)
    }
```

Another versions (for Swift 3 and Swift 2.3) seem to have the same initializer as well.

**My suggestion here:**

To remove this convenience initializer to discourage usage of arbitrary reachability checks without address / or with direct IPv4

**Comment by cnoon** [comment]
Hi @soxjke,

I'm sorry that your use of Alamofire led you to a rejection with Apple. In Alamofire, we've strived to support the same functionality as Apple does in their [Reachability](https://developer.apple.com/library/content/samplecode/Reachability/Introduction/Intro.html) example app which **IS** IPv6 compliant as is clearly called out in their [README](https://developer.apple.com/library/content/samplecode/Reachability/Listings/ReadMe_md.html#//apple_ref/doc/uid/DTS40007324-ReadMe_md-DontLinkElementID_11). There's a lot of history around this particular logic and it being IPv6 compliant in #1228.

We've also tried to clearly call out exactly what you're calling out in this ticket in our [README](https://github.com/Alamofire/Alamofire#network-reachability) section covering reachability. It is important for anyone using the Alamofire reachability system to understand how it should be used, and we've tried to do our best with that in the documentation. Unfortunately, we can't "make" everyone read the documentation before they start using Alamofire and that can lead to these types of issues.

If you don't mind me asking, did you read the documentation about reachability before implementing it? If yes, was it not clear? Is there something else we can do better there? We don't want anyone to get rejected, but there are legit uses of the zero address approach so we'd rather not remove it.

Cheers. 🍻
